dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerCommandBuilder.cs

```diff
@@ -373,6 +374,68 @@ internal static SqlCommand SelectVector<TRecord>(
         return command;
     }
 
+    internal static SqlCommand SelectWhere<TRecord>(
+        Expression<Func<TRecord, bool>> filter,
+        int top,
+        FilterOptions<TRecord> options,
+        SqlConnection connection, string? schema, string tableName,
+        VectorStoreRecordModel model,
+        IEnumerable<KeyValuePair<VectorStoreRecordPropertyModel, bool>> orderByProperties)
+    {
+        SqlCommand command = connection.CreateCommand();
+
+        StringBuilder sb = new(200);
+        sb.AppendFormat("SELECT ");
+        sb.AppendColumnNames(model.Properties, includeVectors: options.IncludeVectors);
+        sb.AppendLine();
+        sb.Append("FROM ");
+        sb.AppendTableName(schema, tableName);
+        sb.AppendLine();
+        if (filter is not null)
+        {
+            int startParamIndex = command.Parameters.Count;
+
+            SqlServerFilterTranslator translator = new(model, filter, sb, startParamIndex: startParamIndex);
+            translator.Translate(appendWhere: true);
+            List<object> parameters = translator.ParameterValues;
+
+            foreach (object parameter in parameters)
+            {
+                command.AddParameter(property: null, $"@_{startParamIndex++}", parameter);
+            }
+            sb.AppendLine();
+        }
+        bool orderByClauseNotAdded = true;
+        foreach (var pair in orderByProperties)
+        {
+            if (orderByClauseNotAdded)
+            {
+                sb.Append("ORDER BY ");
+                orderByClauseNotAdded = false;
+            }
+
+            sb.AppendFormat("[{0}] {1},", pair.Key.StorageName, pair.Value ? "ASC" : "DESC");
+        }
+
+        if (!orderByClauseNotAdded)
+        {
+            sb.Length--; // remove the last comma
+            sb.AppendLine();
+        }
+        else
+        {
+            // no order by properties, but we need to add something for OFFSET and NEXT to work
+            sb.AppendLine("ORDER BY (SELECT NULL)");
```

Comment:
I took this idea from https://www.microsoftpressstore.com/articles/article.aspx?p=2314819

Reply:
Here's the EF variant ([source](https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Query/Internal/SqlServerQuerySqlGenerator.cs#L320)):

> "ORDER BY (SELECT 1)"

That's what happens when you try to force your users to do something they don't want to do 🤣